sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -139,9 +139,27 @@ static void hda_stream_put_dma(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu
 	snd_hdac_ext_stream_release(hext_stream, HDAC_EXT_STREAM_TYPE_LINK);
 }
 
+static int hda_dma_setup(struct snd_sof_dev *sdev, void *dma_data, unsigned int val)
+{
+	struct hdac_ext_stream *hext_stream = dma_data;
+
+	snd_hdac_ext_stream_setup(hext_stream, val);
+
+	return 0;
+}
+
+static void hda_dma_free(struct snd_sof_dev *sdev, void *dma_data)
+{
+	struct hdac_ext_stream *hext_stream = dma_data;
+
+	snd_hdac_ext_stream_reset(hext_stream);
+}
+
 static struct snd_sof_dai_dma_ops hda_dma_ops = {
 	.stream_get_dma = hda_stream_get_dma,
 	.stream_put_dma = hda_stream_put_dma,
+	.setup = hda_dma_setup,
+	.free = hda_dma_free,
```

Comment:
commit title should be updated to ASoC: SOF: Intel: hda-dai: Add setup/free DAI DMA ops